service/resource/endpoint/update.go

```diff
@@ -3,32 +3,53 @@ package endpoint
 import (
 	"context"
 
+	"k8s.io/apimachinery/pkg/api/errors"
+
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/framework"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateState interface{}) error {
+	k8sEndpoint, err := toK8sEndpoint(updateState)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	_, err = r.k8sClient.CoreV1().Endpoints(k8sEndpoint.Namespace).Update(k8sEndpoint)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
 func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*framework.Patch, error) {
-	create, err := r.newCreateChange(ctx, obj, currentState, desiredState)
+	endpoint, err := r.newCreateChange(ctx, obj, currentState, desiredState)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	update, err := r.newUpdateChange(ctx, obj, currentState, desiredState)
+	if len(endpoint.IPs) == 0 {
+		// Nothing to do.
+		return nil, nil
+	}
+
+	k8sEndpoint, err := constructK8sEndpoint(endpoint, r.k8sClient)
```

Comment:
Again, this has nothing to do here. The indicator for me that the PR is missing something was that the private `new*Change` methods where not changed. All the logic should be done there.

Reply:
Ack got it, going to fix this.